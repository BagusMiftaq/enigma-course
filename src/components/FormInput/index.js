import { FormControl, FormLabel, FormGroup } from "react-bootstrap";

const FormInput = ({ label, type, placeholder, value, onChange }) => {
  let props;
  switch (type) {
    case "textarea":
      props = { as: type, value };
      break;
    case "file":
      props = { type };
      break;
    default:
      props = { type, value };
      break;
  }

  const onChangeHandler = (e) => {
    if (type === "file") {
      onChange(e.target.files[0]);
    } else {
      onChange(e.target.value);
    }
  };
  return (
    <FormGroup className="mb-3">
      <FormLabel>{label}</FormLabel>
      <FormControl
        {...props}
        placeholder={placeholder}
        onChange={onChangeHandler}
      />
    </FormGroup>
  );
};

export default FormInput;
