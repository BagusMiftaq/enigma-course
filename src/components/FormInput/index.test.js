import {render} from "@testing-library/react";

import FormInput from "./index";

test("should render FormInput correctly",()=>{
    const mockProps = {props:{
        type: "text"
        }}
    const { asFragment } = render(<FormInput {...mockProps}/>);

    expect(asFragment()).toMatchSnapshot();
})