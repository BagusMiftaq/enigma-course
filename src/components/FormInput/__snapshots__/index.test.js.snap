// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FormInput correctly 1`] = `
<DocumentFragment>
  <div
    class="mb-3"
  >
    <label
      class="form-label"
    />
    <input
      class="form-control"
      value=""
    />
  </div>
</DocumentFragment>
`;
